modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -535,6 +535,16 @@ class StreamingInput final: public cv::gimpl::GIslandExecutable::IInput
             // Stop case
             return cv::gimpl::StreamMsg{cv::gimpl::EndOfStream{}};
         }
+        // Wrap all input cv::Mats with RMats
+        for (auto& arg : isl_input_args) {
+            if (arg.index() == cv::GRunArg::index_of<cv::Mat>()) {
+                // FIXME: This whole construct is ugly, from
+                // its writing to a need in this in general
+                arg = cv::GRunArg{ cv::make_rmat<cv::gimpl::RMatAdapter>(cv::util::get<cv::Mat>(arg))
+                                 , arg.meta
+                                 };
+            }
+        }
```

Comment:
Wait, why this code has moved from the base class (== serving for every other backend) to this particular place?

Reply:
...would a regular `GExecutor` work without that?

Reply:
Regular GExecutor wraps cv::Mats in bindInArg and bindOutArg functions. This part makes streaming backend interface consistent, all cv::Mats are now wrapped, without this change run(StreamingInput, StreamingOutput) gets unwrapped cv::Mat